### PR TITLE
Enable or disable the Impersonate feature

### PR DESCRIPTION
**Pull request by R0MANDEV**
#### Type Of Change 
- [x] Small bug fix
- [x] New feature or enhancement
- [ ] Breaking change

#### Checklist:
- [x] Commits are named and have tag following [commit rules](https://github.com/irontec/ivozprovider/blob/bleeding/doc/dev/en/commits.md)
- [x] Commits are split per component (schema, portal/platform, kamusers, agis, ..)
- [ ] Changes have been tested locally
- [ ] Fixes an existing issue (Fixes #XXXX) 
- [ ] Upport from existing Pull request #XXXX

### CODE REVIEW

**Review by ironArt3mis — CHANGES_REQUESTED** [review]
Functional review required

File: web/portal/brand/src/entities/Administrator/Action/Impersonate.tsx [hunk 1]
```diff
@@ -10,13 +10,16 @@ import {
...
 import { useStoreState } from 'store';
 
+import { WebPortalPropertiesList } from '../../../entities/WebPortal/WebPortalProperties';
+
 const Impersonate: ActionFunctionComponent = (props: ActionItemProps) => {
   const { row, entityService, variant = 'icon' } = props;
   const token = useStoreState((state) => state.auth.token);
   const customData = useStoreState((state) => state.list.customData);
+  const profile = useStoreState((state) => state.clientSession.aboutMe.profile);
+  const isCanImpersonate = !profile?.canImpersonate;
```
Comment (danigargar) [inline web/portal/brand/src/entities/Administrator/Action/Impersonate.tsx:22]:
```suggestion
  const canImpersonate = !profile?.canImpersonate;
```

File: web/portal/brand/src/entities/Company/Action/Impersonate.tsx [hunk 2]
```diff
@@ -10,19 +10,24 @@ import {
...
 const Impersonate: ActionFunctionComponent = (props: ActionItemProps) => {
   const { row, entityService, variant = 'icon' } = props;
   const token = useStoreState((state) => state.auth.token);
-  const customData = useStoreState((state) => state.list.customData);
+  const customData = useStoreState((state) => state.list.customData) as
+    | WebPortalPropertyList<string>[]
+    | null;
 
+  const profile = useStoreState((state) => state.clientSession.aboutMe.profile);
+  const isCanImpersonate = !profile?.canImpersonate;
```
Comment (danigargar) [inline web/portal/brand/src/entities/Company/Action/Impersonate.tsx:25]: idem

File: web/portal/platform/src/entities/Administrator/Action/Impersonate.tsx [hunk 3]
```diff
@@ -25,15 +25,16 @@ const Impersonate: ActionFunctionComponent = (props: ActionItemProps) => {
   const customData = useStoreState(
     (state) => state.list.customData
   ) as WebPortalPropertiesList;
-
+  const profile = useStoreState((state) => state.clientSession.aboutMe.profile);
+  const isCanImpersonate = !profile?.canImpersonate;
```
Comment (danigargar) [inline web/portal/platform/src/entities/Administrator/Action/Impersonate.tsx:29]: idem

File: web/portal/platform/src/entities/Brand/Action/Impersonate.tsx [hunk 4]
```diff
@@ -22,12 +22,13 @@ const Impersonate: ActionFunctionComponent = (props: ActionItemProps) => {
   const customData = useStoreState(
     (state) => state.list.customData
   ) as WebPortalPropertiesList;
-
+  const profile = useStoreState((state) => state.clientSession.aboutMe.profile);
+  const isCanImpersonate = !profile?.canImpersonate;
```
Comment (danigargar) [inline web/portal/platform/src/entities/Brand/Action/Impersonate.tsx:26]: idem

**Review by danigargar — CHANGES_REQUESTED** [review]
1) Put cypress tests and pacts on its own commit.
2) Why no tests for platform?

**Comment by R0MANDEV** [comment]
1.- files sorted by commit
2.- No tests were added to the platform as we have separate cards to perform all Cypress tests on the platform.

File: web/portal/brand/src/entities/Company/Action/Impersonate.tsx [hunk 5]
```diff
@@ -10,19 +10,24 @@ import {
...
 const Impersonate: ActionFunctionComponent = (props: ActionItemProps) => {
   const { row, entityService, variant = 'icon' } = props;
   const token = useStoreState((state) => state.auth.token);
-  const customData = useStoreState((state) => state.list.customData);
+  const customData = useStoreState((state) => state.list.customData) as
+    | WebPortalPropertyList<string>[]
+    | null;
 
+  const profile = useStoreState((state) => state.clientSession.aboutMe.profile);
+  const canImpersonate = !profile?.canImpersonate;
```
Comment (danigargar) [inline web/portal/brand/src/entities/Company/Action/Impersonate.tsx:25]: Idem

File: web/portal/platform/src/entities/Brand/Action/Impersonate.tsx [hunk 6]
```diff
@@ -22,12 +22,13 @@ const Impersonate: ActionFunctionComponent = (props: ActionItemProps) => {
   const customData = useStoreState(
     (state) => state.list.customData
   ) as WebPortalPropertiesList;
-
+  const profile = useStoreState((state) => state.clientSession.aboutMe.profile);
+  const canImpersonate = !profile?.canImpersonate;
```
Comment (danigargar) [inline web/portal/platform/src/entities/Brand/Action/Impersonate.tsx:26]: Idem

**Review by danigargar — CHANGES_REQUESTED** [review]
Rename this commit:
[tests: tests: updated cypress pacts brand](https://github.com/irontec/ivozprovider/pull/2768/commits/ea49d9693a94c0cc72c1cffefeb6f675a271dbd6)

I suggest mixing this changes with "portal/brand: update cypress fixtures", but at least remove the extra "tests:".

**Review by danigargar — APPROVED** [review]

**Review by danigargar — APPROVED** [review]

File: web/portal/brand/src/entities/Administrator/Action/Impersonate.tsx [hunk 7]
```diff
@@ -25,16 +27,26 @@ const Impersonate: ActionFunctionComponent = (props: ActionItemProps) => {
...
+              placement='bottom-start'
+              enterTouchDelay={0}
+            >
+              <span>
+                <StyledTableRowCustomCta disabled={true}>
+                  <AdminPanelSettingsIcon />
+                </StyledTableRowCustomCta>
+              </span>
+            </Tooltip>
+          </>
```
Comment (danigargar) [inline web/portal/brand/src/entities/Administrator/Action/Impersonate.tsx:49]:
```suggestion
          <Tooltip
            title={_('Impersonate')}
            placement='bottom-start'
            enterTouchDelay={0}
          >
            <span>
              <StyledTableRowCustomCta disabled={true}>
                <AdminPanelSettingsIcon />
              </StyledTableRowCustomCta>
            </span>
          </Tooltip>
```

...or you will end with <> twice.

File: web/portal/brand/src/entities/Company/Action/Impersonate.tsx [hunk 8]
```diff
@@ -17,21 +17,33 @@ const Impersonate: ActionFunctionComponent = (props: ActionItemProps) => {
...
+              placement='bottom-start'
+              enterTouchDelay={0}
+            >
+              <span>
+                <StyledTableRowCustomCta disabled={true}>
+                  <AdminPanelSettingsIcon />
+                </StyledTableRowCustomCta>
+              </span>
+            </Tooltip>
+          </>
```
Comment (danigargar) [inline web/portal/brand/src/entities/Company/Action/Impersonate.tsx:46]: Idem

File: web/portal/platform/src/entities/Administrator/Action/Impersonate.tsx [hunk 9]
```diff
@@ -25,24 +25,35 @@ const Impersonate: ActionFunctionComponent = (props: ActionItemProps) => {
...
+              placement='bottom-start'
+              enterTouchDelay={0}
+            >
+              <span>
+                <StyledTableRowCustomCta disabled={true}>
+                  <AdminPanelSettingsIcon />
+                </StyledTableRowCustomCta>
+              </span>
+            </Tooltip>
+          </>
```
Comment (danigargar) [inline web/portal/platform/src/entities/Administrator/Action/Impersonate.tsx:56]: Idem

File: web/portal/platform/src/entities/Brand/Action/Impersonate.tsx [hunk 10]
```diff
@@ -22,21 +22,33 @@ const Impersonate: ActionFunctionComponent = (props: ActionItemProps) => {
...
+              placement='bottom-start'
+              enterTouchDelay={0}
+            >
+              <span>
+                <StyledTableRowCustomCta disabled={true}>
+                  <AdminPanelSettingsIcon />
+                </StyledTableRowCustomCta>
+              </span>
+            </Tooltip>
+          </>
```
Comment (danigargar) [inline web/portal/platform/src/entities/Brand/Action/Impersonate.tsx:51]: idem

File: web/portal/brand/cypress/e2e/Clients/VirtualPbxs/VirtualPbxs.cy.js [hunk 11]
```diff
@@ -19,6 +20,23 @@ describe('in Companies', () => {
     cy.get('table').should('contain', CompaniesCollection.body[0].id);
   });
 
+  it('Impersonate enabled', () => {
+    cy.intercept('GET', '**/api/brand/vPbx/1', {
+      ...PBXsItem,
+    }).as('getAdministrator-1');
```
Comment (danigargar) [inline web/portal/brand/cypress/e2e/Clients/VirtualPbxs/VirtualPbxs.cy.js:26]: Why getAdministrator-1? 
It's a VPbx client (or company), not an admin.

**Review by danigargar — CHANGES_REQUESTED** [review]
Please, if possible, add a test for an administrator who can not impersonate.

**Review by danigargar — APPROVED** [review]

**Review by danigargar — APPROVED** [review]

**Pull request merged**